scripts/lib/CIME/utils.py

```diff
@@ -174,12 +174,26 @@ def _read_cime_config_file():
     CIME_MODEL=e3sm,cesm
     PROJECT=someprojectnumber
     """
+    allowed_in_main = ("cime_model", "project", "charge_account", "srcroot", "mail_type",
```

Comment:
Can you also add a check that `[main]` and `[create_test]` are the only config sections? If I have my test settings in `[test]` instead of `[create_test]` I'd like to get an error instead of those settings being silently ignored.